:seedling: Sync with upstream open-cluster-management-io/cluster-permission (Go 1.26) (#307)
stolostron/cluster-permission: CLAUDE.md

- **`Dockerfile.rhtap`** -- RHTAP/Konflux-oriented container build
- **`COMPONENT_NAME`** -- Downstream component naming for builds
- **`rpms.in.yaml`** / **`rpms.lock.yaml`** -- RPM lock inputs for Konflux/cachi2-style builds

> - **`OWNERS`** -- The downstream OWNERS lists the Red Hat ACM team (dhaiducek, tesshuflower, rokej, mikeshng), which is entirely different from the upstream community maintainers. Always preserve the stolostron version during syncs.

Reviewer: (nit) Remove specific usernames for maintainability.

```suggestion
- **`OWNERS`** -- The downstream OWNERS lists the Red Hat ACM team, which is entirely different from the upstream community maintainers. Always preserve the stolostron version during syncs.
```

go mod vendor
```

### Step 5: Verify vendor integrity and build

After resolving all conflicts, run these two commands to catch any remaining
issues (e.g. leftover conflict markers in vendor files that `git add` won't
detect, or files that `go mod vendor` would add/remove):

```bash
# Re-sync vendor directory -- must produce no output (no changes)

> go mod vendor

Reviewer: (Optional) There a make target for this we could leverage

```suggestion
make deps
```

# Confirm the code compiles cleanly

> go build ./...

Reviewer: (Optional) Alternatively, running `make lint` here is effectively the same as running `go build` and would also have the additional benefit of raise linting issues.

```suggestion
make lint
```

```

If `go mod vendor` produces changes, stage and amend them into the merge commit
before continuing. If `go build` fails, trace the error back to the conflicted
file and fix it.

Also verify that no upstream commits were lost in the merge:

```bash
# upstream/main must be a reachable ancestor of your branch HEAD
git merge-base --is-ancestor upstream/main HEAD && echo "OK - all upstream commits preserved" || echo "ERROR - upstream commits missing!"

# Should print nothing (no commits in upstream that aren't in your branch)
git log --oneline upstream/main --not HEAD
```

### Step 6: Update the RHTAP Dockerfile

If the upstream sync includes a **Go version bump** (check `go.mod` for a changed `go` directive),
the RHTAP Dockerfile must be updated manually -- it is downstream-only and won't be touched by
the merge.

```bash
# Check current Go version in go.mod

> head -5 go.mod

Reviewer: (nit) There's a command for this 🙂 

```suggestion
go list -m -json go
```

# Check current version in the RHTAP Dockerfile

> grep "golang-builder" Dockerfile.rhtap

Reviewer: (Optional) Maybe use something more generic in case the image changes?

```suggestion
grep -i "^FROM .* AS builder$" Dockerfile.rhtap
```

```

> Update `Dockerfile.rhtap` to match (e.g. `rhel_9_1.25` -> `rhel_9_1.26`):

Reviewer: (nit) Wording update

```suggestion
Update the builder FROM image tag in `Dockerfile.rhtap` to match the Go version in `go.mod` (e.g. `rhel_9_1.25` -> `rhel_9_1.26`):
```

> The builder image tag pattern is `rhel_9_<goversion>`, e.g.:

Reviewer: (nit) This is eventually going to be RHEL 10, so maybe

```suggestion
The builder image tag pattern is `rhel_<rhel-version>_<go-version>`, e.g.:
```

```
FROM brew.registry.redhat.io/rh-osbs/openshift-golang-builder:rhel_9_1.26 AS builder
```

Note: the runtime base image (`ubi9/ubi-minimal`) and Red Hat `LABEL` blocks in `Dockerfile.rhtap`
are intentionally different from the upstream Dockerfile and should **not** be changed to match

> upstream.

Reviewer: It's interesting the `Dockerfile.rhtap` made it into upstream. We should probably instead delete that at some point since it's only relevant for Red Hat builds and just leave this note out once it's removed.

Author: I've added another comment in the CLAUDE.md to check if the upstream changes involve changes to the Dockerfile, and check if any need to be pulled into Dockerfile.rhap - hope that will help if we remove the rhtap from the upstream.
